### PR TITLE
Ref

### LINKED ISSUE #23: delete a cart item

### LINKED ISSUE #59: start documentation

### LINKED ISSUE #60: separate server routes into different files

### PR DESCRIPTION
**Pull request by quiyetbrul**

### CODE REVIEW

**Comment by quiyetbrul** [comment]
handlebars in this branch arent updating... cant test front end properly...

**Comment by quiyetbrul** [comment]
PLS DO NOT MERGE. I JUST SAW MORE UPDATES ON THE SERVER.JS

File: controllers/cart.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  const cart = await Cart.findOne({ where: { UserUsername: user.username } });
+  const items = await Cart.getItems();
+
+  const data = {
+    user: user,
+    cart: cart,
+    items:items,
+  };
+
+  res.status(200).render('homepage', { data });
```
Comment (dannyreina) [inline controllers/cart.js]: render cart page

File: controllers/item.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+//  DELETE ITEM FUNCTIONALITY  //
+///////////////////////////////
+// delete an item
+exports.deleteItem = async (req, res) => {
+  const user = await User.findByPk(req.params.username);
+  const item = await Item.findByPk(req.params.id);
+
+  await item.destroy();
+
+  res.status(200).redirect(`/users/${user.username}`);
```
Comment (dannyreina) [inline controllers/item.js]: apply 301 code

File: controllers/user.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+		password: newPassword,
+	});
+
+	// create new cart and assign it to new user
+	const newUserCart = await Cart.create({
+		totalPrice: 0,
+		UserUsername: newUser.username,
+	});
+
+	res.status(200).redirect(`/homepage/${newUser.username}`);
```
Comment (dannyreina) [inline controllers/user.js]: apply 301 code

File: controllers/user.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+	await user.set({
+		username: udpatedUsername === '' ? user.username : udpatedUsername,
+		fullName: updatedFullName === '' ? user.fullName : updatedFullName,
+		email: updatedEmail === '' ? user.email : updatedEmail,
+		password: updatedPassword === '' ? user.password : updatedPassword,
+		isAdmin: updatedIsAdmin === '' ? user.isAdmin : updatedIsAdmin,
+	});
+
+	await user.save();
+	res.status(200).redirect(`/users/${user.username}`);
```
Comment (dannyreina) [inline controllers/user.js]: apply 301 code

**Review by dannyreina — CHANGES_REQUESTED** [review]

**Pull request merged**